src/hotspot/share/services/attachListener.cpp

```diff
@@ -127,9 +127,11 @@ class attachStream : public bufferedStream {
   // Called after the operation is completed.
   // If reply_writer is provided, writes the results.
   void complete() {
-    JavaThread* thread = JavaThread::current();
-    ThreadBlockInVM tbivm(thread);
-    flush_reply();
+    if (_reply_writer != nullptr) {
```

Comment:
Is this sufficient? I'm looking at JDK-8319055 for the first time and I can't figure out how the value of `_reply_writer` affects the `_allow_streaming` logic that accesses the reply-writer. We seem to enable streaming mode in platform independent code, yet the availability of the reply-writer is platform dependent! ???

Reply:
Ah found it - sorry:
```
_allow_streaming(reply_writer == nullptr ? false : allow_streaming),
```